reladomographql/src/main/java/com/gs/reladomo/graphql/SchemaProvider.java

```diff
@@ -0,0 +1,111 @@
+
+    private SchemaProvider ()
+    {
+    }
+
+    public static GraphQLSchema forResource (String schemaResourceName)
+    {
+        URL url = Resources.getResource (schemaResourceName);
+        LOGGER.info ("Loading from " + url.getPath ());
+        URL metaUrl = Resources.getResource ("meta.graphqls");
+        String metaSdl = null;
+        try
+        {
+            metaSdl = Resources.toString (metaUrl, Charsets.UTF_8);
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException ("Failed to read " + metaUrl, e);
+        }
+
+        try
+        {
+            String sdl = Resources.toString (url, Charsets.UTF_8) + "\n" + metaSdl;
+            GraphQLSchema graphQLSchema = new SchemaProvider ().buildSchema (sdl);
+            GraphQL.newGraphQL (graphQLSchema).build ();
+
+            return graphQLSchema;
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException ("Failed to read config from " + url, e);
+        }
+    }
+
+    /**
+     * see: https://www.graphql-java.com/documentation/v12/schema/
+     */
+    private GraphQLSchema buildSchema (String sdl)
+    {
+        TypeDefinitionRegistry typeRegistry = new graphql.schema.idl.SchemaParser ().parse (sdl);
+        RuntimeWiring runtimeWiring = buildWiring ();
+        SchemaGenerator schemaGenerator = new SchemaGenerator ();
+        return schemaGenerator.makeExecutableSchema (typeRegistry, runtimeWiring);
+    }
+
+    private RuntimeWiring buildWiring ()
+    {
+        RuntimeWiring.Builder builder = RuntimeWiring.newRuntimeWiring ();
+        builder = builder.scalar (ExtendedScalars.DateTime).scalar (ExtendedScalars.Date).scalar (ExtendedScalars.Url).scalar (new TimestampScalar());
+
+        for (MithraRuntimeCacheController each : MithraManagerProvider.getMithraManager ().getRuntimeCacheControllerSet ())
+        {
+            String className = each.getClassName ();
+            String name = StringUtils.decapitalize (className, className.lastIndexOf ('.') + 1);
+            LOGGER.info ("building " + name + " for " + className);
+
+            builder = builder
+                    .type (newTypeWiring ("Mutation")
+                            .dataFetcher (name + "_insert", new ReladomoMutationFetcher (name, each.getClassName (), each.getFinderInstance ())))
+                    .type (newTypeWiring ("Query")
+                            .dataFetcher (name + "ById", new ReladomoQueryFetcher (each.getFinderInstance (), false)))
+                    .type (newTypeWiring ("Query")
+                            .dataFetcher (name + "s", new ReladomoQueryFetcher (each.getFinderInstance (), true)))
+                    .type (newTypeWiring ("Query")
+                            .dataFetcher (name + "_aggregate", new AggregateQueryFetcher (each.getFinderInstance ())))
+                    .type (newTypeWiring ("Query"));
+
+        }
+
+        return builder.build ();

```

Comment:
If I understand this loop correctly, there are many copies of the type named Query, each with a single data fetcher. java-graphql must have logic in it to merge all these types into one?

I thought it was required to call type() just once for each type name.